climada/hazard/centroids/centr.py

```diff
@@ -172,6 +177,40 @@ def from_base_grid(land=False, res_as=360, base_file=None):
             centroids = centroids.select(reg_id=land_reg_ids)
         return centroids
 
+    @staticmethod
+    def from_geodataframe(gdf):
+        """Create centroids instance from GeoDataFrame. All columns beside the
+        geometry (or geom) column will be translated using pd.Series.to_numpy,
+        so the Series dtype will be respected.
+
+        >>> gdf = geopandas.read_file('centroids.shp')
+        >>> gdf.region_id = gdf.region_id.astype(int)  # type coercion
+        >>> centroids = Centroids.from_geodataframe(gdf)
+
+        Parameters:
+            gdf (GeoDataFrame): Where the geometry column needs to consist of
+                point features.
+        """
+        centroids = Centroids()
+
+        centroids.geometry = gdf.geometry
+        centroids.lat = gdf.geometry.y.to_numpy()
```

Comment:
consider `to_numpy(copy=True)`

Reply:
AFAIK, the `copy=False` does not necessarily mean that the result is _not_ copied. I think I'll just trust the memory manager to decide sensibly. Maybe it even copies lazily as soon as the derived objects actually get manipulated?

Reply:
I haven't got the time to learn about such details of python memory management, but what you're describing sounds scary. I'm just going to put in the mandatory copy operation, even if it may be less efficient.